python/lsst/resources/_resourceHandles/_httpResourceHandle.py

```diff
@@ -63,7 +65,9 @@ def fileno(self) -> int:
         raise io.UnsupportedOperation("HttpReadResourceHandle does not have a file number")
 
     def flush(self) -> None:
-        raise io.UnsupportedOperation("HttpReadResourceHandles are read only")
+        modes = set(self._mode.split())
+        if {"w", "x", "a", "+"} & modes:
+            raise io.UnsupportedOperation("HttpReadResourceHandles are read only")
```

Comment:
This doesn't look right, although maybe `self._mode` isn't a normal Python mode string? `split()` splits on whitespace, so a mode string of `w+` or `wb` would test as false in this test, but it looks like the intent is for it to be true.

Reply:
I love it when tests pass without a problem despite the code being completely wrong. I see that `modes = set(self._mode)` does work just fine...